core/src/main/scala/cats/syntax/either.scala

```diff
@@ -1,13 +1,163 @@
+    }
+
+  def leftMap[C](f: A => C): Either[C, B] = eab match {
+    case Left(a)      => Left(f(a))
+    case r @ Right(_) => r.asInstanceOf[Either[C, B]]
+  }
+
+  def flatMap[AA >: A, D](f: B => Either[AA, D]): Either[AA, D] = eab match {
+    case l @ Left(_) => l.asInstanceOf[Either[AA, D]]
+    case Right(b)    => f(b)
+  }
+
+  def compare[AA >: A, BB >: B](that: Either[AA, BB])(implicit AA: Order[AA], BB: Order[BB]): Int = eab.fold(
+    a => that.fold(AA.compare(a, _), _ => -1),
+    b => that.fold(_ => 1, BB.compare(b, _))
+  )
+
+  def partialCompare[AA >: A, BB >: B](that: Either[AA, BB])(implicit AA: PartialOrder[AA], BB: PartialOrder[BB]): Double =
+    eab.fold(
+      a => that.fold(AA.partialCompare(a, _), _ => -1),
+      b => that.fold(_ => 1, BB.partialCompare(b, _))
+    )
+
+  def ===[AA >: A, BB >: B](that: Either[AA, BB])(implicit AA: Eq[AA], BB: Eq[BB]): Boolean = eab.fold(
+    a => that.fold(AA.eqv(a, _), _ => false),
+    b => that.fold(_ => false, BB.eqv(b, _))
+  )
+
+  def traverse[F[_], AA >: A, D](f: B => F[D])(implicit F: Applicative[F]): F[Either[AA, D]] = eab match {
+    case l @ Left(_) => F.pure(l.asInstanceOf[Either[AA, D]])
+    case Right(b) => F.map(f(b))(Right(_))
+  }
+
+  def foldLeft[C](c: C)(f: (C, B) => C): C = eab.fold(_ => c, f(c, _))
+
+  def foldRight[C](lc: Eval[C])(f: (B, Eval[C]) => Eval[C]): Eval[C] =
+    eab.fold(_ => lc, b => f(b, lc))
+
+  /**
+   * Combine with another `Either` value.
+   *
+   * If this `Either` is a `Left` then it will be returned as-is.
+   * If this `Either` is a `Right` and `that` `Either` is a left, then `that` will be
+   * returned.
+   * If both `Either`s are `Right`s, then the `Semigroup[BB]` instance will be used
+   * to combine both values and return them as a `Right`.
+   * Note: If both `Either`s are `Left`s then their values are not combined. Use
+   * `Validated` if you prefer to combine `Left` values.
+   *
+   * Examples:
+   * {{{
+   * scala> import cats.implicits._
+   * scala> val l1: Either[String, Int] = Either.left("error 1")
+   * scala> val l2: Either[String, Int] = Either.left("error 2")
+   * scala> val r3: Either[String, Int] = Either.right(3)
+   * scala> val r4: Either[String, Int] = Either.right(4)
+   *
+   * scala> l1 combine l2
+   * res0: Either[String, Int] = Left(error 1)
+   *
+   * scala> l1 combine r3
+   * res1: Either[String, Int] = Left(error 1)
+   *
+   * scala> r3 combine l1
+   * res2: Either[String, Int] = Left(error 1)
+   *
+   * scala> r3 combine r4
+   * res3: Either[String, Int] = Right(7)
+   * }}}
+   */
+  final def combine[AA >: A, BB >: B](that: Either[AA, BB])(implicit BB: Semigroup[BB]): Either[AA, BB] = eab match {
+    case left @ Left(_) => left
+    case Right(b1) => that match {
+      case left @ Left(_) => left
+      case Right(b2) => Right(BB.combine(b1, b2))
+    }
+  }
+
+  def show[AA >: A, BB >: B](implicit AA: Show[AA], BB: Show[BB]): String = eab.fold(
```

Comment:
Looks like we don't need it, I'll check around for other uses and remove them if possible.